wgengine/router/ifconfig_windows.go

```diff
@@ -574,26 +574,21 @@ func deltaNets(a, b []*net.IPNet) (add, del []*net.IPNet) {
 	return
 }
 
-func excludeIPv6LinkLocal(in []*net.IPNet) (out []*net.IPNet) {
-	out = in[:0]
-	for _, n := range in {
-		if len(n.IP) == 16 && n.IP.IsLinkLocalUnicast() {
-			// Windows creates a fixed link-local address for wintun,
-			// which doesn't seem to route correctly. Unfortunately, LLMNR returns this
-			// address for lookups by the hostname, and Windows prefers using it.
-			// This means that local traffic addressed to the machine's hostname breaks.
-			//
-			// While we otherwise preserve link-local addresses, we delete
-			// this one to force lookups to use a working address.
-			//
-			// See: https://github.com/tailscale/tailscale/issues/4647
-			if ip, ok := netaddr.FromStdIP(n.IP); !ok || wintunLinkLocal != ip {
-				continue // filter this IPNet
-			}
+func isIPv6LinkLocal(in *net.IPNet) bool {
+	return len(in.IP) == 16 && in.IP.IsLinkLocalUnicast()
+}
+
+func sliceSplit[T any](in []T, f func(T) bool) (a, b []T) {
+	a = in[:0]
+	b = in[:0]
+	for _, v := range in {
+		if f(v) {
+			a = append(a, v)
+		} else {
+			b = append(b, v)
```

Comment:
this doesn't seem right. they're aliasing the same memory.


Also doesn't seem like the right package for it.

Reply:
e.g. https://go.dev/play/p/B3HiKEthA45

Reply:
yeah, not surprised, copilot wrote this code